flowmachine/flowmachine/features/subscriber/visit_count_filtered_reference_location.py

```diff
@@ -0,0 +1,61 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# -*- coding: utf-8 -*-
+import numpy as np
+from typing import List, Optional
+
+from flowmachine.core.query import Query
+from flowmachine.features.subscriber.location_visits import LocationVisits
+from flowmachine.features.utilities.subscriber_locations import BaseLocation
+
+
+class VisitCountFilteredReferenceLocation(Query, BaseLocation):
+    """
+    A filtered reference location, which includes rows only where the subscriber
+    visited the location a specified number of times.
+
+    Parameters
+    ----------
+    reference_locations_query : BaseLocation
+        Reference location to take locations from
+    location_visits_query : LocationVisits
+        Location visit counts to use for thresholding
+    lower_bound : int
+        Minimum number of visits to allow a location to be included
+    upper_bound : int, default Inf
+        Maximum number of visits to allow a location to be included. Defaults to unbounded.
+    """
+
+    def __init__(
+        self,
+        reference_locations_query: BaseLocation,
+        location_visits_query: LocationVisits,
```

Comment:
Does this need to be a `LocationVisits` query? E.g. I think this would work equally well with `CallDays` - or any query with (subscriber, location, value) columns.